src/platforms/esp32/components/avm_builtins/network_driver.c

```diff
@@ -19,157 +19,335 @@
  * SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
  */
 
+#include <atom.h>
+#include <context.h>
+#include <debug.h>
+#include <globalcontext.h>
+#include <interop.h>
+#include <mailbox.h>
+#include <platform_defaultatoms.h>
+#include <port.h>
 #include <sdkconfig.h>
-#ifdef CONFIG_AVM_ENABLE_NETWORK_PORT_DRIVER
+#include <term.h>
+#include <utils.h>
+
+// #define ENABLE_TRACE 1
+#include <trace.h>
 
-#include "port.h"
+#include <esp32_sys.h>
+#include <esp_log.h>
+#include <esp_wifi.h>
+#include <lwip/inet.h>
 
 #include <string.h>
 
-#include "atom.h"
-#include "context.h"
-#include "debug.h"
-#include "globalcontext.h"
-#include "interop.h"
-#include "mailbox.h"
-#include "port.h"
-#include "term.h"
-#include "utils.h"
+#include <esp_sntp.h>
 
-#include "esp32_sys.h"
+#define TCPIP_HOSTNAME_MAX_SIZE 255
 
-#include "platform_defaultatoms.h"
+#define TAG "network_driver"
+#define PORT_REPLY_SIZE TUPLE_SIZE(2) + REF_SIZE
```

Comment:
Let's use parenthesis also here.
If for any reason I do `PORT_REPLY_SIZE * 2` it gets replaced with `TUPLE_SIZE(2) + REF_SIZE * 2` which gives very bad headaches.